fiftyone/core/dataset.py

```diff
@@ -4511,6 +4511,19 @@ def _create_indexes(sample_collection_name, frame_collection_name):
         )
 
 
+def _declare_fields(doc_cls, field_docs):
+    for field_name, field in doc_cls._fields.items():
+        if isinstance(field, fof.EmbeddedDocumentField):
+            doc = foo.SampleFieldDocument.from_field(field)
+            field = doc.to_field()
+            field._set_parent(doc_cls)
+            doc_cls._fields[field_name] = field
+            setattr(doc_cls, field_name, field)
+
+    for field_doc in field_docs or []:
+        doc_cls._declare_field(field_doc.to_field())
```

Comment:
When creating sample and frame backing document classes, default emedded document field instances, e.g. `Metadata`, must be explicitly recreated and populated with dynamic fields. If they are not recreated, then the default instance is polluted and future datasets would load with dynamic fields on other datasets.